app/controllers/candidate_interface/course_choices/course_selection_controller.rb

```diff
@@ -0,0 +1,79 @@
+module CandidateInterface
+  module CourseChoices
+    class CourseSelectionController < BaseController
+      def options_for_course
+        if params[:course_choice_id]
+          @course_choice_id = params[:course_choice_id]
+          current_application_choice = current_application.application_choices.find(@course_choice_id)
+
+          @pick_course = PickCourseForm.new(
+            provider_id: params.fetch(:provider_id),
+            application_form: current_application,
+            course_id: current_application_choice.course.id,
+          )
+        else
+          @pick_course = PickCourseForm.new(
+            provider_id: params.fetch(:provider_id),
+            application_form: current_application,
+          )
+        end
+      end
+
+      def pick_course
```

Comment:
This method still looks quite long, but it's now in a much shorter file. This is a lot better and likely enough for now.